### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: shared/idmap/shift_linux.go [hunk 1]
```diff
@@ -280,18 +280,63 @@ static int get_userns_fd_cb(void *data)
...
+		return -errno;
+
+	// Setup gid_map
+	snprintf(path, sizeof(path), "/proc/%d/gid_map", pid);
+	fd = openat(AT_FDCWD, path, O_WRONLY);
+	if (fd < 0)
+		return -errno;
+
+	if (write(fd, "0 0 1", 5) != 5)
+		return -errno;
```
Comment (hallyn) [inline shared/idmap/shift_linux.go:334]: Note that since this is c and not go, fd will not be cleaned up in this error path.

**Review by hallyn — CHANGES_REQUESTED** [review]
(looks good otherwise)

File: shared/idmap/shift_linux.go (hunk 1 above)
Comment (stgraber) [reply shared/idmap/shift_linux.go:339]: Fixed!

**Comment by stgraber** [comment]
Added close calls to the write error handling path, that should do it.

**Review by hallyn — APPROVED** [review]

File: shared/idmap/shift_linux.go [hunk 2]
```diff
@@ -279,19 +279,72 @@ static int get_userns_fd_cb(void *data)
...
+	// Setup gid_map
+	snprintf(path, sizeof(path), "/proc/%d/gid_map", pid);
+	file_fd = openat(AT_FDCWD, path, O_WRONLY);
+	if (file_fd < 0)
+		goto err_process;
+
+	if (write(file_fd, "0 0 1", 5) != 5)
+		goto err_process;
+
+	if (close(file_fd) < 0)
```
Comment (hallyn) [inline shared/idmap/shift_linux.go:331]: If you close file_fd here, you should set it back to -EBADF before potentially closing it a second time.

Comment (stgraber) [reply shared/idmap/shift_linux.go]: Done!

**Review by hallyn — APPROVED** [review]

**Pull request merged**